No rule to make target '/opt/nodemcu-firmware/sdk/esp32-esp-idf/make/project.mk'.  Stop.

### Expected behavior
`make menuconfig` works

### Actual behavior
```text
2020-01-03 07:20:52 (55.1 MB/s) - '/opt/nodemcu-firmware/cache/toolchain-esp32-linux-x86_64-20181106.1.tar.xz' saved [28337296/28337296]

# the archive contains ro files
Setting IDF_PATH and re-invoking...
Makefile:42: /opt/nodemcu-firmware/sdk/esp32-esp-idf/make/project.mk: No such file or directory
make[1]: Entering directory '/opt/nodemcu-firmware'
make[1]: *** No rule to make target '/opt/nodemcu-firmware/sdk/esp32-esp-idf/make/project.mk'.  Stop.
make[1]: Leaving directory '/opt/nodemcu-firmware'
Makefile:18: recipe for target 'menuconfig' failed
make: *** [menuconfig] Error 2
```

### Test code
https://github.com/mk-pmb/nodemcu-firmware-daily-vanilla/runs/372007960

### NodeMCU version
branch dev-esp32, commit bfd2259

### Hardware
The Gitlab cloud

--- a/Makefile
+++ b/Makefile
@@ -1,3 +1,5 @@
+# -*- coding: utf-8, tab-width: 2 -*-
+
 # If we haven't got IDF_PATH set already, reinvoke with IDF_PATH in the environ
 .NOTPARALLEL:
 
@@ -19,6 +21,8 @@ all: | $(ESP32_GCC)
 	@env IDF_PATH=$(IDF_PATH) PATH=$(PATH):$(ESP32_BIN) $(MAKE) -f $(THIS_MK_FILE) $@
 	@if test "$@" = "clean"; then rm -rf $(THIS_DIR)/tools/toolchains/esp32-*; fi
 
+install_toolchain: $(ESP32_GCC)
+
 $(ESP32_GCC): $(ESP32_TOOLCHAIN_DL)
 	@echo Uncompressing toolchain
 	@mkdir -p $(THIS_DIR)/tools/toolchains/
@@ -26,6 +30,8 @@ $(ESP32_GCC): $(ESP32_TOOLCHAIN_DL)
 	# the archive contains ro files
 	@chmod -R u+w $(THIS_DIR)/tools/toolchains/esp32-*
 	@touch $@
+
+download_toolchain: $(ESP32_TOOLCHAIN_DL)
 
 $(ESP32_TOOLCHAIN_DL):
 	@mkdir -p $(THIS_DIR)/cache
@@ -38,6 +44,15 @@ PROJECT_NAME:=NodeMCU
 # This is, sadly, the cleanest way to resolve the different non-standard
 # conventions for sized integers across the various components.
 BASIC_TYPES=-Du32_t=uint32_t -Du16_t=uint16_t -Du8_t=uint8_t -Ds32_t=int32_t -Ds16_t=int16_t -Duint32=uint32_t -Duint16=uint16_t -Duint8=uint8_t -Dsint32=int32_t -Dsint16=int16_t -Dsint8=int8_t
+
+$(IDF_PATH)/make/project.mk:
+	@echo "ESP-IDF isn't properly installed, will try to re-install and then re-run make:"
+	git submodule init
+	git submodule update --recursive
+	python -m pip install --user --requirement $(IDF_PATH)/requirements.txt || echo "W: pip failed, but this might be harmless. Let's just try whether it works anyway." >&2
+  # If we'd just continue, make would fake-succeed albeit with a warning
+  # "make[1]: *** No rule to make target `menuconfig'.  Stop."
+#	$(MAKE) -f $(THIS_MK_FILE) $@ || exit $?
 
 include $(IDF_PATH)/make/project.mk
 

